src/CachePlugin.php

```diff
@@ -195,13 +231,78 @@ private function getMaxAge(ResponseInterface $response)
     private function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults([
+            'cache_lifetime' => 86400 * 30, // 30 days
             'default_ttl' => null,
             'respect_cache_headers' => true,
             'hash_algo' => 'sha1',
         ]);
 
+        $resolver->setAllowedTypes('cache_lifetime', 'int');
         $resolver->setAllowedTypes('default_ttl', ['int', 'null']);
         $resolver->setAllowedTypes('respect_cache_headers', 'bool');
         $resolver->setAllowedValues('hash_algo', hash_algos());
     }
+
+    /**
+     * @param CacheItemInterface $cacheItem
+     *
+     * @return ResponseInterface
+     */
+    private function createResponseFromCacheItem(CacheItemInterface $cacheItem)
+    {
+        $data = $cacheItem->get();
+
+        /** @var ResponseInterface $response */
+        $response = $data['response'];
+        $response = $response->withBody($this->streamFactory->createStream($data['body']));
+
+        return $response;
+    }
+
+    /**
+     * Get the value of the "If-Modified-Since" header.
+     *
+     * @param CacheItemInterface $cacheItem
+     *
+     * @return string|null
+     */
+    private function getModifiedSinceHeaderValue(CacheItemInterface $cacheItem)
+    {
+        $data = $cacheItem->get();
+        if (!isset($data['createdAt'])) {
+            return;
+        }
+
+        $modified = new \DateTime('@'.$data['createdAt']);
+        $modified->setTimezone(new \DateTimeZone('GMT'));
+
+        return sprintf('%s GMT', $modified->format('l, d-M-y H:i:s'));
+    }
+
+    /**
+     * Get the ETag from the cached response.
+     *
+     * @param CacheItemInterface $cacheItem
+     *
+     * @return string|null
+     */
+    private function getETag(CacheItemInterface $cacheItem)
+    {
+        $data = $cacheItem->get();
+        if (!isset($data['etag'])) {
+            return;
+        }
+
+        if (is_array($data['etag'])) {
```

Comment:
Maybe negate this and return the bottom thing to avoid the nesting, and also would avoid needing that extra return